components/brave_ads/core/internal/ads/serving/eligible_ads/pipelines/notification_ads/eligible_notification_ads_v3.cc

```diff
@@ -0,0 +1,148 @@
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/ads/serving/eligible_ads/pipelines/notification_ads/eligible_notification_ads_v3.h"
+
+#include <utility>
+
+#include "absl/types/optional.h"
+#include "base/functional/bind.h"
+#include "brave/components/brave_ads/core/internal/ads/ad_events/ad_events_database_table.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/choose/predict_ad_embeddings.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/eligible_ads/exclusion_rules/exclusion_rules_util.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/eligible_ads/exclusion_rules/notification_ads/notification_ad_exclusion_rules.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/serving_features.h"
+#include "brave/components/brave_ads/core/internal/ads/serving/targeting/user_model_info.h"
+#include "brave/components/brave_ads/core/internal/ads_client_helper.h"
+#include "brave/components/brave_ads/core/internal/common/logging_util.h"
+#include "brave/components/brave_ads/core/internal/creatives/notification_ads/creative_notification_ads_database_table.h"
+#include "brave/components/brave_ads/core/internal/geographic/subdivision/subdivision_targeting.h"
+#include "brave/components/brave_ads/core/internal/resources/behavioral/anti_targeting/anti_targeting_resource.h"
+
+namespace ads::notification_ads {
+
+EligibleAdsV3::EligibleAdsV3(
+    geographic::SubdivisionTargeting* subdivision_targeting,
+    resource::AntiTargeting* anti_targeting_resource)
+    : EligibleAdsBase(subdivision_targeting, anti_targeting_resource) {}
+
+void EligibleAdsV3::GetForUserModel(
+    targeting::UserModelInfo user_model,
+    GetEligibleAdsCallback<CreativeNotificationAdList> callback) {
+  BLOG(1, "Get eligible notification ads");
+
+  database::table::AdEvents database_table;
+  database_table.GetForType(
+      mojom::AdType::kNotificationAd,
+      base::BindOnce(&EligibleAdsV3::OnGetForUserModel, base::Unretained(this),
+                     std::move(user_model), std::move(callback)));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void EligibleAdsV3::OnGetForUserModel(
+    targeting::UserModelInfo user_model,
+    GetEligibleAdsCallback<CreativeNotificationAdList> callback,
+    const bool success,
+    const AdEventList& ad_events) {
+  if (!success) {
+    BLOG(1, "Failed to get ad events");
+    std::move(callback).Run(/*had_opportunity*/ false, {});
+    return;
+  }
+
+  GetBrowsingHistory(std::move(user_model), ad_events, std::move(callback));
+}
+
+void EligibleAdsV3::GetBrowsingHistory(
+    targeting::UserModelInfo user_model,
+    const AdEventList& ad_events,
+    GetEligibleAdsCallback<CreativeNotificationAdList> callback) {
+  const int max_count = features::GetBrowsingHistoryMaxCount();
+  const int days_ago = features::GetBrowsingHistoryDaysAgo();
```

Comment:
nit: rather than assigning we can just move to the GetBrowsingHistory call (please see other eligible_*_ads_v* files

Reply:
I am not seeing the difference with the other servings

Reply:
This change is under code review in other servings, and will be merged soon.